### PR TITLE
Make MainViz the renderer of templates, fix DonutChart bugz, upgrade packages

### PR DESCRIPTION
**Pull request by ianks**
This makes MainViz responsible for creating the proper template components to be rendered. This will be useful as it the first level of seperation b/t the templates and the controller. At this point, the templates have no knowledge about how to retrieve data, as the data is set in the templates props. Yaya!

Fun fact of the day: ES6 arrow funcs also bind `this`!

``` js
( )=> { return this.foo}

// is the same as

function() { return this.foo}.bind(this)
```

### CODE REVIEW

File: app/components/template-viewer/TemplateViewer.jsx [hunk 1]
```diff
@@ -12,19 +12,19 @@ class TemplateViewer extends React.Component {
...
-        }.bind(this));
+    constructor(props) {
+      super(props);
+      this.state = {templates: []}
+
+      $.getJSON("blobs/templates.json", (data) => {
+        this.setState({templates: data});
+      }.bind(this)).fail((error) => {
+        console.log(error);
+      }.bind(this));
```
Comment (develra) [inline app/components/template-viewer/TemplateViewer.jsx]: Are the .bind(this)'s redundant based on our new knowledge about arrow functions?

File: app/components/visualizer/main-viz/main-viz.jsx [hunk 2]
```diff
@@ -1,3 +1,26 @@
 import DonutChartComponent from "./templates/DonutChart/DonutChartComponent.jsx";
 
-React.render(<DonutChartComponent />, document.getElementById("main-viz"));
+class MainViz extends React.Component {
+  constructor(props) {
+    super(props);
```
Comment (develra) [inline app/components/visualizer/main-viz/main-viz.jsx:5]: Could you explain the super(props); - I'm just curious about what it does.

**Comment by develra** [comment]
Good overall! :+1:

File: app/components/template-viewer/TemplateViewer.jsx (hunk 1 above)
Comment (ianks) [inline app/components/template-viewer/TemplateViewer.jsx]: yes, the bind is totally redundant. i will clean this up.

File: app/components/visualizer/main-viz/main-viz.jsx (hunk 2 above)
Comment (ianks) [inline app/components/visualizer/main-viz/main-viz.jsx:5]: `super(props)` calls the parent constructor (`React.Component`) function. This initializes the component.

**Pull request merged**